### PR TITLE
Remove global_step argument

### PR DESCRIPTION
**Pull request by tagomatech**
Following discussion in issue [#113](https://github.com/tensorflow/agents/issues/113), replace  `self._optimizer.apply_gradients(grads_and_vars, global_step=self.train_step_counter)` with     `self._optimizer.apply_gradients(grads_and_vars)` to ensure TF2 compatibility.

### CODE REVIEW

**Comment by tagomatech** [comment]
@googlebot I signed it!

**Comment by tagomatech** [comment]
@googlebot I signed it!

**Comment by tagomatech** [comment]
@googlebot I signed it!

File: tf_agents/agents/dqn/dqn_agent.py [hunk 1]
```diff
@@ -294,8 +294,7 @@ def _train(self, experience, weights):
       eager_utils.add_gradients_summaries(grads_and_vars,
                                           self.train_step_counter)
 
-    self._optimizer.apply_gradients(grads_and_vars,
-                                    global_step=self.train_step_counter)
+    self._optimizer.apply_gradients(grads_and_vars)
```
Comment (sguada) [inline tf_agents/agents/dqn/dqn_agent.py:372]: add

```
# Increment train_step_counter
self.train_step_counter.assign_add(1)
```

**Review by sguada — CHANGES_REQUESTED** [review]

File: tf_agents/agents/dqn/dqn_agent.py (hunk 1 above)
Comment (tagomatech) [reply tf_agents/agents/dqn/dqn_agent.py:372]: You are perfectly right @sguada. Thank you. Done.

**Comment by sguada** [comment]
Sorry there was a change in this file, so you need to sync to head and merge the changes.

**Comment by ebrevdo** [comment]
I thought we have a PR that selectively passes global_step iff the optimizer is a v1 optimizer.

**Comment by tagomatech** [comment]
> Sorry there was a change in this file, so you need to sync to head and merge the changes.

The file has changed substantially, indeed. I made changes requested. Hopefully I did it the right way. Thank you for all the work and patience.

File: tf_agents/agents/dqn/dqn_agent.py [hunk 2]
```diff
@@ -14,9 +14,7 @@
 # limitations under the License.
 
 """A DQN Agent.
-
```
Comment (sguada) [inline tf_agents/agents/dqn/dqn_agent.py:17]: Undo all the space removal.

File: tf_agents/agents/dqn/dqn_agent.py [hunk 3]
```diff
@@ -364,8 +349,10 @@ def _train(self, experience, weights):
       eager_utils.add_gradients_summaries(grads_and_vars,
                                           self.train_step_counter)
     training.apply_gradients(self._optimizer,
-                             grads_and_vars,
```
Comment (sguada) [inline tf_agents/agents/dqn/dqn_agent.py:370]: replace all this with
```

self._optimizer.apply_gradients(grads_and_vars)

# Increment train_step_counter
self.train_step_counter.assign_add(1)
```

**Review by sguada — CHANGES_REQUESTED** [review]
Undo all the space removal in the file.

File: tf_agents/agents/dqn/dqn_agent.py (hunk 2 above)
Comment (tagomatech) [reply tf_agents/agents/dqn/dqn_agent.py:17]: You were right, several spaces disappeared. Thank you.

File: tf_agents/agents/dqn/dqn_agent.py (hunk 3 above)
Comment (tagomatech) [reply tf_agents/agents/dqn/dqn_agent.py:370]: Apologies for the delay. I just hadn't seen your new replacement request. Thank you.

**Review by sguada — CHANGES_REQUESTED** [review]
Can you revert all the changes you made un-related to the the global_step change?

**Comment by tfboyd** [comment]
I feel bad this is the 3rd PR closed you authored.  You did a bunch of commits and worked to address the issues.  We have gotten much better the past 3 months at accepting PRs.  This one looks to be stale as the underlying code to apply the gradients looks to be using a utility class now.  Your solution was correct, we failed to process it fast enough.  Super sorry and we are getting better.

**Comment by tagomatech** [comment]
no worries. thanks! and thanks for TF agents.
